app/support/reservations/duration_change_validations.rb

```diff
@@ -0,0 +1,32 @@
+class Reservations::DurationChangeValidations
+  include ActiveModel::Validations
+
+  validate :start_time_not_changed, unless: Proc.new { |r| r.reservation.reserve_start_at_editable? }
+  validate :duration_not_shortened, if: Proc.new { |r| Time.zone.now >= r.reservation.reserve_start_at }
+
+  def initialize(reservation)
+    @reservation = reservation
+  end
+
+  def reservation
+    @reservation
+  end
+
+  def copy_errors!
+    errors.each do |e|
+      reservation.errors.add(e)
+    end
+  end
+
+  def start_time_not_changed
+    if reservation.reserve_start_at_changed?
```

Comment:
Would it make sense to move the check we're doing in the Proc into here? Would that be easier to read?